Realm/Realm.Weaver/Extensions/TypeReferenceExtensions.cs

```diff
@@ -46,6 +49,76 @@ SequencePoint GetPropSequencePoint()
         }
     }
 
+    public static bool IsRealmObjectDescendant(this TypeReference @this, ImportedReferences references) =>
+        IsDescendantOf(@this, references.RealmObject, references.EmbeddedObject, references.AsymmetricObject);
+
+    public static bool IsAsymmetricObjectDescendant(this TypeReference @this, ImportedReferences references) =>
+        IsDescendantOf(@this, references.AsymmetricObject);
+
+    public static bool IsEmbeddedObjectDescendant(this TypeReference @this, ImportedReferences references) =>
+        IsDescendantOf(@this, references.EmbeddedObject);
+
+    public static bool IsSameAs(this TypeReference @this, TypeReference other)
+    {
+        if (@this is null || other is null)
+        {
+            return false;
+        }
+
+        return @this.FullName == other.FullName && @this.GetAssemblyName() == other.GetAssemblyName();
+    }
+
+    public static string GetAssemblyName(this TypeReference @this)
+    {
+        switch (@this.Scope.MetadataScopeType)
+        {
+            case MetadataScopeType.AssemblyNameReference:
+                return ((AssemblyNameReference)@this.Scope).FullName;
+            case MetadataScopeType.ModuleReference:
+                return ((ModuleReference)@this.Scope).Name;
+            default:
+                return ((ModuleDefinition)@this.Scope).Assembly.FullName;
+        }
+    }
+
+    public static string ToFriendlyString(this TypeReference type)
```

Comment:
Not sure I'm super happy about all these reorderings - perhaps they make sense, but I would much rather we do them in a separate PR as it inflates the number of lines everyone needs to go through and there's no clear indication what changes are real and what are just moving code around.

Reply:
I'll avoid this in the future. I've added comments everywhere it's not new code but just shuffled around.